### ISSUE TITLE
BlobTrigger memory leak

### ISSUE THREAD

**Issue by jaykay-design** [Azure/azure-webjobs-sdk#562]
I'm listening with a BlobTrigger for any new/updated blobs.

```
public static void BackupBlobsMedia([BlobTrigger("media")] ICloudBlob sourceBlob)
  {
        BackupBlobs(sourceBlob);
  }
```

The container is quite large (thousands of blobs).

While there are no blobs being updated looking at the process explorer I see a linear increase of memory allocated until the WebJob has allocated all of it and everything slows down to almost a stop.

The heap dump shows large amounts of Blob* and Uri objects.

285068      6841632 Microsoft.Azure.WebJobs.Host.Storage.Blob.StorageBlobProperties
290610      9299520 Microsoft.WindowsAzure.Storage.StorageUri
285068     11402720 Microsoft.Azure.WebJobs.Host.Storage.Blob.StorageBlockBlob
290098     18566272 Microsoft.WindowsAzure.Storage.Blob.CloudBlockBlob
291067     18628288 System.Uri+MoreInfo
470763     19697846      Free
290098     20887056 Microsoft.WindowsAzure.Storage.Blob.BlobAttributes
290943     23275440 System.Collections.Generic.Dictionary`2[[System.String, mscorlib],...
290098     41774112 Microsoft.WindowsAzure.Storage.Blob.BlobProperties
581151     41842872 System.Uri+UriInfo
581153     41843016 System.Uri
2097468    275780538 System.String

Could it be that the initial blob polling algorithm is not designed for large container sizes?

**Comment by mathewc** [comment]
I believe the behavior you're seeing is due to a bug we fixed in the BlobTrigger startup scan logic - previously the scan wasn't following continuation tokens correctly (see [this commit](https://github.com/Azure/azure-webjobs-sdk/commit/2b82dafd098ca5f4a8d97db7899b5e84992c28c3)). Can you try the latest pre-release alpha2 package and confirm that you no longer see the problem?

**Comment by jaykay-design** [comment]
Thanks. I confirm the issues is resolved by using 1.1.0-alpha2.
